File: src/main/java/pe/tarjetaestilos/models/Promotion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

import jakarta.persistence.*;
import java.time.ZonedDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.*;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import com.fasterxml.jackson.annotation.JsonBackReference;
import org.hibernate.annotations.Where;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import java.util.ArrayList;
import java.util.List;
/**
 *
 * @author julito
 */

@Entity
@Table(name="promotions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class Promotion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    private String image;
    
    private String logo;
    
    @Column(name = "promotion_link")
    private String promotion_link;
    
    @Column(name = "promotion_map")
    private String promotion_map;
    
    @Column(name = "outstanding", nullable = true)
    private Boolean outstanding;
    
    @Column(name = "featured", nullable = true)
    private Boolean featured;
    
    @Column(name = "isnew", nullable = true)
    private Boolean isnew;
    
    private String terms_conditions;
    
    private String discount;
        
    @Column(name = "home", nullable = true)
    private Boolean home;
    
    private Boolean exclusive;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;
    
    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;
    
    @ManyToOne()
    @JoinColumn(name = "menu_id", nullable = false)
    @ToString.Exclude
    @JsonBackReference
    private Menu menu;
    
    @ManyToOne()
    @JoinColumn(name = "department_id", nullable = false)
    @ToString.Exclude
    private Department department;
    
    @ManyToOne()
    @JoinColumn(name = "affiliates_id", nullable = false)
    @ToString.Exclude
    private Affiliates affiliates;
    
    @OneToMany(mappedBy = "promotion")
    @JsonManagedReference
    @ToString.Exclude
    private List<PromotionsCategories> promotionsCategories = new ArrayList<>();
    
}

File: src/models/Promotion.ts
class Promotion{
    title:string|null;
    description:string|null;
    image:string|null;
    discount:string|null;
    isnew:boolean|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    department_id:number|null;
    categoryIds:number[]|null;
    id:number|null;

    featured:boolean|null;
    promotion_link:string|null;
    promotion_map:string|null;
    outstanding:boolean|null;
    terms_conditions:string|null;

    affiliates_id:number|null;
    menu_id:number|null;
    home:boolean|null;
    exclusive:boolean|null;

    constructor(
        title:string|null,
        description:string|null,
        image:string|null,
        discount:string|null,
        isnew:boolean|null,
        created_at:Date|null, 
        updated_at:Date|null, 
        deleted_at:Date|null,
        department_id:number|null,
        categoryIds:number[]|null=null, 
        id:number|null=null,
        outstanding:boolean|null=null,
        featured:boolean|null=null,
        promotion_link:string|null=null,
        promotion_map:string|null=null,
        terms_conditions:string|null=null,
        affiliates_id:number|null=null,
        menu_id:number|null=null,
        home:boolean|null=null,
        exclusive:boolean|null=null,
    ){
        this.title = title;
        this.description = description;
        this.image = image;
        this.discount = discount;
        this.isnew = isnew;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.department_id = department_id;
        this.categoryIds = categoryIds;
        this.id = id;
        this.outstanding = outstanding;
        this.featured = featured;
        this.promotion_link = promotion_link;
        this.promotion_map = promotion_map;
        this.terms_conditions = terms_conditions;
        this.affiliates_id = affiliates_id;
        this.menu_id = menu_id;
        this.home = home;
        this.exclusive = exclusive;
    }
};
export default Promotion;